packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -245,6 +257,47 @@ export class ChainService implements ChainServiceInterface {
     return transactionResponse;
   }
 
+  async challenge(
+    challengeStates: SignedState[],
+    privateKey: PrivateKey
```

Comment:
Probably not something we want or need to do now, but perhaps worth considering and making an issue for — it's not great from a security perspective to be passing around this private key in-memory too often. If we split out the `ChainService` into a separate process, or even on a different host, then this will lead to the key being passed over the network. Maybe the challenge message could be something that is easily sign-able within the `Store` for example, maybe?

Reply:
Thanks for flagging this https://github.com/statechannels/statechannels/issues/3044